keps/sig-network/4762-allow-arbitrary-fqdn-as-pod-hostname/README.md

```diff
@@ -0,0 +1,573 @@
+|  3 | `aa`        | `bb`         |                      |                     | `aa`                           | `aa.bb.<ns>.svc.<zone>`        | `aa.bb.<ns>.svc.<zone>`        |
+|  4 |             |              | true                 |                     | `<podname>`                    | `<podname>`                    |                                |
+|  5 | `aa`        |              | true                 |                     | `aa`                           | `aa`                           |                                |
+|  6 |             | `bb`         | true                 |                     | `<podname>.bb.<ns>.svc.<zone>` | `<podname>.bb.<ns>.svc.<zone>` | `<podname>.bb.<ns>.svc.<zone>` |
+|  7 | `aa`        | `bb`         | true                 |                     | `aa.bb.<ns>.svc.<zone>`        | `aa.bb.<ns>.svc.<zone>`        | `aa.bb.<ns>.svc.<zone>`        |
+|  8 |             |              |                      | `xx.yy.zz`          | `xx.yy.zz`                     | `xx.yy.zz`                     |                                |
+|  9 | `aa`        |              |                      | `xx.yy.zz`          | `xx.yy.zz`                     | `xx.yy.zz`                     |                                |
+| 10 |             | `bb`         |                      | `xx.yy.zz`          | `xx.yy.zz`                     | `xx.yy.zz`                     | `<podname>.bb.<ns>.svc.<zone>` |
+| 11 | `aa`        | `bb`         |                      | `xx.yy.zz`          | `xx.yy.zz`                     | `xx.yy.zz`                     | `aa.bb.<ns>.svc.<zone>`        |
+| 12 |             |              | true                 | `xx.yy.zz`          | INVALID                        | INVALID                        | INVALID                        |
+| 13 | `aa`        |              | true                 | `xx.yy.zz`          | INVALID                        | INVALID                        | INVALID                        |
+| 14 |             | `bb`         | true                 | `xx.yy.zz`          | INVALID                        | INVALID                        | INVALID                        |
+| 15 | `aa`        | `bb`         | true                 | `xx.yy.zz`          | INVALID                        | INVALID                        | INVALID                        |
+
+
+
+As shown in the table, setting `hostnameOverride` will only change the hostname inside the pod and will not modify the DNS records in Kubernetes.
+
+### Test Plan
+
+##### Prerequisite testing updates
+
+##### Unit tests
+
+- Add kubelet unit tests to verify that container hostnames are correctly generated:  `k8s.io/kubernetes/pkg/kubelet/kuberuntime`: `2025-06-06` - `69.0%`
+- Add API validation unit tests to ensure all field combinations yield correct results: `k8s.io/kubernetes/pkg/apis/core/validation` : `2025-06-06` - `84.7%`
+
+##### Integration tests
+
+- N/A
+
+##### e2e tests
```

Comment:
should this field be a part of conformance under the `test/e2e`? The test may add some new methods in agnhost to check what the pod think it is.

Reply:
/hold cancel

I was under the impression it was going to merge, sorry about  the hold ... but we should have a matrix of e2e tests that validates the table in https://github.com/kubernetes/enhancements/pull/4768#discussion_r2136481743 , we already have most of them, so it will be more building the table